In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

### Pre existing data of crimes

1.District wise crimes committed

In [2]:
di = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\01_District_wise_crimes_committed_IPC_2001_2012.csv")
di

STATE/UT       DISTRICT  YEAR  MURDER  ATTEMPT TO MURDER  \
0     ANDHRA PRADESH       ADILABAD  2001     101                 60   
1     ANDHRA PRADESH      ANANTAPUR  2001     151                125   
2     ANDHRA PRADESH       CHITTOOR  2001     101                 57   
3     ANDHRA PRADESH       CUDDAPAH  2001      80                 53   
4     ANDHRA PRADESH  EAST GODAVARI  2001      82                 67   
...              ...            ...   ...     ...                ...   
9012     LAKSHADWEEP    LAKSHADWEEP  2012       0                  0   
9013     LAKSHADWEEP          TOTAL  2012       0                  0   
9014      PUDUCHERRY       KARAIKAL  2012       5                  6   
9015      PUDUCHERRY     PUDUCHERRY  2012      24                 21   
9016      PUDUCHERRY          TOTAL  2012      29                 27   

      CULPABLE HOMICIDE NOT AMOUNTING TO MURDER  RAPE  CUSTODIAL RAPE  \
0                                            17    50               0   
1                                             1    23               0   
2                                             2    27               0   
3                                             1    20               0   
4                                             1    23               0   
...                                         ...   ...             ...   
9012                                          0     0               0   
9013                                          0     0               0   
9014                                          2     6               0   
9015                                         10     7               0   
9016                                         12    13               0   

      OTHER RAPE  KIDNAPPING & ABDUCTION  ...  ARSON  HURT/GREVIOUS HURT  \
0             50                      46  ...     30                1131   
1             23                      53  ...     69                1543   
2             27                      59  ...     38                2088   
3             20                      25  ...     23                 795   
4             23                      49  ...     41                1244   
...          ...                     ...  ...    ...                 ...   
9012           0                       0  ...      3                   3   
9013           0                       0  ...      3                   3   
9014           6                       2  ...      1                 186   
9015           7                      17  ...     20                 632   
9016          13                      19  ...     21                 818   

      DOWRY DEATHS  ASSAULT ON WOMEN WITH INTENT TO OUTRAGE HER MODESTY  \
0               16                                                149     
1                7                                                118     
2               14                                                112     
3               17                                                126     
4               12                                                109     
...            ...                                                ...     
9012             0                                                  1     
9013             0                                                  1     
9014             0                                                  2     
9015             0                                                  7     
9016             0                                                  9     

      INSULT TO MODESTY OF WOMEN  CRUELTY BY HUSBAND OR HIS RELATIVES  \
0                             34                                  175   
1                             24                                  154   
2                             83                                  186   
3                             38                                   57   
4                             58                                  247   
...               

In [3]:
di.isnull().sum().sum()

0

2.District wise crimes agaiinst SC

In [4]:
sc = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\02_01_District_wise_crimes_committed_against_SC_2001_2012.csv")
sc

STATE/UT       DISTRICT  Year  Murder  Rape  \
0     ANDHRA PRADESH       ADILABAD  2001       0     1   
1     ANDHRA PRADESH      ANANTAPUR  2001       0     4   
2     ANDHRA PRADESH       CHITTOOR  2001       3     3   
3     ANDHRA PRADESH       CUDDAPAH  2001       0     3   
4     ANDHRA PRADESH  EAST GODAVARI  2001       1     3   
...              ...            ...   ...     ...   ...   
9013     LAKSHADWEEP    LAKSHADWEEP  2012       0     0   
9014     LAKSHADWEEP          TOTAL  2012       0     0   
9015      PUDUCHERRY       KARAIKAL  2012       1     0   
9016      PUDUCHERRY     PUDUCHERRY  2012       1     0   
9017      PUDUCHERRY          TOTAL  2012       2     0   

      Kidnapping and Abduction  Dacoity  Robbery  Arson  Hurt  \
0                            4        0        0      0     3   
1                            0        0        0      0    49   
2                            0        0        0      0    38   
3                            0        0        0      0    20   
4                            0        0        0      0     3   
...                        ...      ...      ...    ...   ...   
9013                         0        0        0      0     0   
9014                         0        0        0      0     0   
9015                         0        0        0      0     0   
9016                         0        0        0      0     0   
9017                         0        0        0      0     0   

      Prevention of atrocities (POA) Act  \
0                                      0   
1                                     21   
2                                     36   
3                                     52   
4                                     12   
...                                  ...   
9013                                   0   
9014                                   0   
9015                                   0   
9016                                   1   
9017                                   1   

      Protection of Civil Rights (PCR) Act  Other Crimes Against SCs  
0                                       15                        32  
1                                        0                        53  
2                                        0                        34  
3                                        0                        25  
4                                       63                         7  
...                                    ...                       ...  
9013                                     0                         0  
9014                                     0                         0  
9015                                     0                         0  
9016                                    20                         1  
9017                                    20                         1  

[9018 rows x 13 columns]

In [5]:
sc.isnull().sum().sum()

0

3.District wise crimes against St

In [6]:
st = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\02_District_wise_crimes_committed_against_ST_2001_2012.csv")
st

STATE/UT       DISTRICT  Year  Murder  Rape  Kidnapping Abduction  \
0     ANDHRA PRADESH       ADILABAD  2001       0     1                     2   
1     ANDHRA PRADESH      ANANTAPUR  2001       0     0                     0   
2     ANDHRA PRADESH       CHITTOOR  2001       0     0                     0   
3     ANDHRA PRADESH       CUDDAPAH  2001       0     0                     0   
4     ANDHRA PRADESH  EAST GODAVARI  2001       0     0                     0   
...              ...            ...   ...     ...   ...                   ...   
9013     LAKSHADWEEP    LAKSHADWEEP  2012       0     0                     0   
9014     LAKSHADWEEP          TOTAL  2012       0     0                     0   
9015      PUDUCHERRY       KARAIKAL  2012       0     0                     0   
9016      PUDUCHERRY     PUDUCHERRY  2012       0     0                     0   
9017      PUDUCHERRY          TOTAL  2012       0     0                     0   

      Dacoity  Robbery  Arson  Hurt  Protection of Civil Rights (PCR) Act  \
0           0        0      0     2                                     0   
1           0        0      0     7                                     0   
2           0        0      0     2                                     0   
3           0        0      0     2                                     0   
4           0        0      0     0                                     0   
...       ...      ...    ...   ...                                   ...   
9013        0        0      0     0                                     0   
9014        0        0      0     0                                     0   
9015        0        0      0     0                                     0   
9016        0        0      0     0                                     0   
9017        0        0      0     0                                     0   

      Prevention of atrocities (POA) Act  Other Crimes Against STs  
0                                      0                        13  
1                                      1                         6  
2                                      0                         0  
3                                      2                         0  
4                                      0                        14  
...                                  ...                       ...  
9013                                   0                         0  
9014                                   0                         0  
9015                                   0                         0  
9016                                   0                         0  
9017                                   0                         0  

[9018 rows x 13 columns]

4. District wise crimes against children

In [7]:
ch = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\03_District_wise_crimes_committed_against_children_2001_2012.csv")
ch

STATE/UT       DISTRICT  Year  Murder  Rape  \
0     ANDHRA PRADESH       ADILABAD  2001     0.0   0.0   
1     ANDHRA PRADESH      ANANTAPUR  2001    19.0  12.0   
2     ANDHRA PRADESH       CHITTOOR  2001     0.0   0.0   
3     ANDHRA PRADESH       CUDDAPAH  2001     0.0   0.0   
4     ANDHRA PRADESH  EAST GODAVARI  2001     0.0   0.0   
...              ...            ...   ...     ...   ...   
9010     LAKSHADWEEP    LAKSHADWEEP  2012     0.0   0.0   
9011     LAKSHADWEEP          TOTAL  2012     0.0   0.0   
9012      PUDUCHERRY       KARAIKAL  2012     0.0   3.0   
9013      PUDUCHERRY     PUDUCHERRY  2012     0.0   4.0   
9014      PUDUCHERRY          TOTAL  2012     0.0   7.0   

      Kidnapping and Abduction  Foeticide  Abetment of suicide  \
0                          0.0        0.0                  0.0   
1                         29.0        0.0                  6.0   
2                          0.0        0.0                  0.0   
3                          0.0        0.0                  0.0   
4                          0.0        0.0                  0.0   
...                        ...        ...                  ...   
9010                       0.0        0.0                  0.0   
9011                       0.0        0.0                  0.0   
9012                       2.0        0.0                  0.0   
9013                      16.0        0.0                  0.0   
9014                      18.0        0.0                  0.0   

      Exposure and abandonment  Procuration of minor girls  \
0                          0.0                         0.0   
1                          0.0                         0.0   
2                          0.0                         0.0   
3                          0.0                         0.0   
4                          0.0                         0.0   
...                        ...                         ...   
9010                       0.0                         0.0   
9011                       0.0                         0.0   
9012                       1.0                         0.0   
9013                       2.0                         0.0   
9014                       3.0                         0.0   

      Buying of girls for prostitution  Selling of girls for prostitution  \
0                                  0.0                                0.0   
1                                  0.0                                0.0   
2                                  0.0                                0.0   
3                                  0.0                                0.0   
4                                  0.0                                0.0   
...                                ...                                ...   
9010                               0.0                                0.0   
9011                               0.0                                0.0   
9012                               0.0                                0.0   
9013                               0.0                                0.0   
9014                               0.0                                0.0   

      Prohibition of child marriage act  Other Crimes  Total  
0                                   0.0           0.0      0  
1                                   0.0           0.0     66  
2                                   0.0           0.0      0  
3                                   0.0           0.0      0  
4                                   0.0           0.0      0  
...                                 ...           ...    ...  
9010                                0.0           0.0      0  
9011                                0.0           0.0      0  
9012                                0.0           1.0      7  
9013                                1.0           2.0     25  
9014                                1.0           3.0     32  

[9015 rows x 15 columns]

5. Juvenile apprehanded for the crime state wise

In [8]:
ju = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\08_01_Juvenile_apprehended_state_IPC_2001_2010.csv")
ju

STATE/UT  Year                                     CRIME  \
0      Andhra Pradesh  2001                                    Murder   
1      Andhra Pradesh  2001                  Attempt to Commit Murder   
2      Andhra Pradesh  2001               C H Not amounting to Murder   
3      Andhra Pradesh  2001                                      Rape   
4      Andhra Pradesh  2001                            Custodial Rape   
...               ...   ...                                       ...   
10495      Puducherry  2010           Cruelty by Husband or Relatives   
10496      Puducherry  2010  Importation of girls from Foreign Countr   
10497      Puducherry  2010               Causing death by negligence   
10498      Puducherry  2010                          Other IPC Crimes   
10499      Puducherry  2010                          Total IPC Crimes   

       Boys 7-12 Years  Girls 7-12 Years  Boys 12-16 Years  Girls 12-16 Years  \
0                    3                 0                 7                  0   
1                    2                 0                 0                  0   
2                    0                 0                 0                  0   
3                    2                 0                15                  0   
4                    0                 0                 0                  0   
...                ...               ...               ...                ...   
10495                0                 0                 0                  0   
10496                0                 0                 0                  0   
10497                0                 0                 0                  0   
10498                0                 0                 1                  0   
10499                0                 0                 8                  0   

       Boys 16-18 Years  Girls 16-18 Years  Total for boys all Age Groups  \
0                     5                  0                             15   
1                    11                  0                             13   
2                     0                  0                              0   
3                     2                  1                             19   
4                     0                  0                              0   
...                 ...                ...                            ...   
10495                 0                  0                              0   
10496                 0                  0                              0   
10497                 0                  0                              0   
10498                 2                  3                              3   
10499                31                  4                             39   

       Total for girls all Age Groups  Grand total  
0                                   0           15  
1                                   0           13  
2                                   0            0  
3                                   1           20  
4                                   0            0  
...                               ...          ...  
10495                               0            0  
10496                               0            0  
10497                               0            0  
10498                               3            6  
10499                               4           43  

[10500 rows x 12 columns]

6. State wise property stolen and recovered nature

In [9]:
ps = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\11_Property_stolen_and_recovered_nature_of_property_2001_2010.csv")
ps

Area_Name  Year              Group_Name  \
0     Andaman & Nicobar Islands  2001       Cattle - Property   
1                Andhra Pradesh  2001       Cattle - Property   
2             Arunachal Pradesh  2001       Cattle - Property   
3                         Assam  2001       Cattle - Property   
4                         Bihar  2001       Cattle - Property   
...                         ...   ...                     ...   
4545                 Tamil Nadu  2010  Total Property(Nature)   
4546                    Tripura  2010  Total Property(Nature)   
4547              Uttar Pradesh  2010  Total Property(Nature)   
4548                Uttarakhand  2010  Total Property(Nature)   
4549                West Bengal  2010  Total Property(Nature)   

                         Sub_Group_Name  Cases_Property_Recovered  \
0                             2. Cattle                         0   
1                             2. Cattle                       448   
2                             2. Cattle                        22   
3                             2. Cattle                       149   
4                             2. Cattle                       144   
...                                 ...                       ...   
4545  Total Property Stolen & Recovered                     16125   
4546  Total Property Stolen & Recovered                       192   
4547  Total Property Stolen & Recovered                      9130   
4548  Total Property Stolen & Recovered                       964   
4549  Total Property Stolen & Recovered                      4548   

      Cases_Property_Stolen  Value_of_Property_Recovered  \
0                         1                            0   
1                       580                      6490596   
2                        34                       135500   
3                       322                       683350   
4                       334                       896019   
...                     ...                          ...   
4545                  21509                    660311804   
4546                    879                      5666102   
4547                  35068                    577591772   
4548                   2234                     47135685   
4549                  23759                   1168242161   

      Value_of_Property_Stolen  
0                         1000  
1                      7233876  
2                       704500  
3                      1816386  
4                      1911068  
...                        ...  
4545                1317919190  
4546                  33032746  
4547                1442670414  
4548                 123398840  
4549                5015168687  

[4550 rows x 8 columns]

7. State wise police killed and injured

In [10]:
pk = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\13_Police_killed_or_injured_on_duty2001_2010.csv")
pk

Area_Name  Year                         Group_Name  \
0     Andaman & Nicobar Islands  2001  Police - Assistant Sub-Inspectors   
1                Andhra Pradesh  2001  Police - Assistant Sub-Inspectors   
2             Arunachal Pradesh  2001  Police - Assistant Sub-Inspectors   
3                         Assam  2001  Police - Assistant Sub-Inspectors   
4                         Bihar  2001  Police - Assistant Sub-Inspectors   
...                         ...   ...                                ...   
2445                 Tamil Nadu  2010                     Police - Total   
2446                    Tripura  2010                     Police - Total   
2447              Uttar Pradesh  2010                     Police - Total   
2448                Uttarakhand  2010                     Police - Total   
2449                West Bengal  2010                     Police - Total   

                         Sub_Group_Name  Police_Injured_By_Criminals  \
0            3. Assistant Sub-Inspectos                            0   
1            3. Assistant Sub-Inspectos                            0   
2            3. Assistant Sub-Inspectos                            0   
3            3. Assistant Sub-Inspectos                            0   
4            3. Assistant Sub-Inspectos                            1   
...                                 ...                          ...   
2445  7. Total Police Killed or Injured                           10   
2446  7. Total Police Killed or Injured                            0   
2447  7. Total Police Killed or Injured                            6   
2448  7. Total Police Killed or Injured                            0   
2449  7. Total Police Killed or Injured                           17   

      Police_Injured_By_Riotous_Mobs  Police_Injured_In_Accidents  \
0                                  0                            0   
1                                  3                            4   
2                                  0                            0   
3                                  0                            0   
4                                  0                            0   
...                              ...                          ...   
2445                              26                          142   
2446                               0                            0   
2447                              10                            9   
2448                               0                            5   
2449                             265                           86   

      Police_Injured_In_Dacoity_OperationsOther_raids  \
0                                                   0   
1                                                   1   
2                                                   0   
3                                                   0   
4                                                   0   
...                                               ...   
2445                                                4   
2446                                                0   
2447                                               10   
2448                                                0   
2449                                               23   

      Police_Injured_In_TerroristsExtremists_Operations  \
0                                                     0   
1                                                     3   
2                                                     0   
3                                                     1   
4                                                     2   
...                                                 ...   
2445                                                  0   
2446                                                  2   
2447                                                  0   
2448                                                  0   
2449                                                 21   

      Police_Injured_

8. State wise case reported

In [11]:
cr = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\17_Case_reported_and_value_of_property_taken_away_by_place_of_occurrence_2001_2012.csv")
cr

STATE/UT  YEAR         Place Of Occurrence  \
0     Andhra Pradesh  2001        RESIDENTIAL PREMISES   
1     Andhra Pradesh  2001                    HIGH-WAY   
2     Andhra Pradesh  2001                 RIVER & SEA   
3     Andhra Pradesh  2001                    RAILWAYS   
4     Andhra Pradesh  2001              RUNNING TRAINS   
...              ...   ...                         ...   
4339      Puducherry  2012                       BANKS   
4340      Puducherry  2012   COMMERCIAL ESTABLISHMENTS   
4341      Puducherry  2012                OTHER PLACES   
4342      Puducherry  2012              TOTAL (MANUAL)   
4343      Puducherry  2012  TOTAL COMPUTER FROM 1 TO 7   

      Dacoity (Section 395-398 IPC) - Number of cases registered  \
0                                                   100            
1                                                    57            
2                                                     2            
3                                                     8            
4                                                     5            
...                                                 ...            
4339                                                  0            
4340                                                  2            
4341                                                  2            
4342                                                  5            
4343                                                  5            

      Dacoity (Section 395-398 IPC) - Value Of Property Stolen (in rupees)  \
0                                               4446961                      
1                                               5340335                      
2                                                145345                      
3                                               1750800                      
4                                                 75000                      
...                                                 ...                      
4339                                                  0                      
4340                                              58000                      
4341                                             575000                      
4342                                             683000                      
4343                                             683000                      

      Robbery(Section 392-394, 397, 398 IPC) - Number of cases registered  \
0                                                   177                     
1                                                   172                     
2                                                    11                     
3                                                    19                     
4                                                     3                     
...                                                 ...                     
4339                                                  0                     
4340                                                  4                     
4341                                                  8                     
4342                                                 16                     
4343                                                 16                     

      Robbery(Section 392-394, 397, 398 IPC) - Value Of Property Stolen (in rupees)  \
0                                               5962460                               
1                                               6364866                               
2                                                209330                               
3                                                304336                               
4                                                164000                               
...                                                 ...                               
4339                       

9. State wise crime by place of occurance

In [12]:
po = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\17_Crime_by_place_of_occurrence_2001_2012.csv")
po


STATE/UT  YEAR  RESIDENTIAL PREMISES - Dacoity  \
0       ANDHRA PRADESH  2001                             100   
1    ARUNACHAL PRADESH  2001                               9   
2                ASSAM  2001                             381   
3                BIHAR  2001                             818   
4         CHHATTISGARH  2001                              54   
..                 ...   ...                             ...   
451              DELHI  2012                              10   
452        LAKSHADWEEP  2012                               0   
453         PUDUCHERRY  2012                               1   
454        TOTAL (UTs)  2012                              11   
455  TOTAL (ALL-INDIA)  2012                            1288   

     RESIDENTIAL PREMISES - Robbery  RESIDENTIAL PREMISES - Burglary  \
0                               177                             5158   
1                                26                               99   
2                               191                             1695   
3                               326                             2486   
4                                42                             3336   
..                              ...                              ...   
451                             101                             1249   
452                               0                                3   
453                               4                               47   
454                             123                             1515   
455                            2821                            60698   

     RESIDENTIAL PREMISES - Theft  HIGHWAYS - Dacoity  HIGHWAYS - Robbery  \
0                            4257                  57                 172   
1                             131                   0                   0   
2                            2901                  46                 136   
3                            4741                 162                 826   
4                            1417                  10                  38   
..                            ...                 ...                 ...   
451                          7076                   1                  32   
452                             7                   0                   0   
453                            42                   0                   0   
454                          7974                   1                  32   
455                         81861                 838                5000   

     HIGHWAYS - Burglary  HIGHWAYS - Theft  ...  \
0                     31                74  ...   
1                      0                 8  ...   
2                      7                87  ...   
3                      0               257  ...   
4                     12                72  ...   
..                   ...               ...  ...   
451                    0               189  ...   
452                    0                 0  ...   
453                    0                 0  ...   
454                    0               199  ...   
455                  316             27802  ...   

     COMMERCIAL ESTABLISHMENTS - Burglary  COMMERCIAL ESTABLISHMENTS - Theft  \
0                                    1041                               2502   
1                                      84                                 54   
2                                     442                                967   
3                                     231                                686   
4                                     370                                299   
..                                    ...                                ...   
451                                   285                               1370   
452                                     0                                  0   
453                                    26                                 20   
454                                   412 

10. State wise Human rights violations data

In [13]:
hr = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\35_Human_rights_violation_by_police_2001_2010.csv")
hr

Area_Name  Year                     Group_Name  \
0        Andhra Pradesh  2001    HR_Disappearance of Persons   
1     Arunachal Pradesh  2001    HR_Disappearance of Persons   
2                 Assam  2001    HR_Disappearance of Persons   
3                 Bihar  2001    HR_Disappearance of Persons   
4            Chandigarh  2001    HR_Disappearance of Persons   
...                 ...   ...                            ...   
2262          Meghalaya  2010  HR_Total Violations by Police   
2263           Nagaland  2010  HR_Total Violations by Police   
2264             Odisha  2010  HR_Total Violations by Police   
2265      Uttar Pradesh  2010  HR_Total Violations by Police   
2266        West Bengal  2010  HR_Total Violations by Police   

                     Sub_Group_Name  \
0      01. Disappearance of Persons   
1      01. Disappearance of Persons   
2      01. Disappearance of Persons   
3      01. Disappearance of Persons   
4      01. Disappearance of Persons   
...                             ...   
2262  12. Total (Sum of 1-11 Above)   
2263  12. Total (Sum of 1-11 Above)   
2264  12. Total (Sum of 1-11 Above)   
2265  12. Total (Sum of 1-11 Above)   
2266  12. Total (Sum of 1-11 Above)   

      Cases_Registered_under_Human_Rights_Violations  Policemen_Chargesheeted  \
0                                                0.0                      0.0   
1                                                0.0                      0.0   
2                                                0.0                      0.0   
3                                                0.0                      0.0   
4                                                0.0                      0.0   
...                                              ...                      ...   
2262                                             0.0                      0.0   
2263                                             0.0                      0.0   
2264                                             5.0                      1.0   
2265                                             6.0                      5.0   
2266                                             1.0                      0.0   

      Policemen_Convicted  
0                     0.0  
1                     0.0  
2                     0.0  
3                     0.0  
4                     0.0  
...                   ...  
2262                  0.0  
2263                  0.0  
2264                  0.0  
2265                  0.0  
2266                  0.0  

[2267 rows x 7 columns]

11. Disrict wise crimes against women

In [14]:
wm = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\India_Crime - Copy\42_District_wise_crimes_committed_against_women_2001_2012.csv")
wm

STATE/UT       DISTRICT  Year  Rape  Kidnapping and Abduction  \
0     ANDHRA PRADESH       ADILABAD  2001    50                        30   
1     ANDHRA PRADESH      ANANTAPUR  2001    23                        30   
2     ANDHRA PRADESH       CHITTOOR  2001    27                        34   
3     ANDHRA PRADESH       CUDDAPAH  2001    20                        20   
4     ANDHRA PRADESH  EAST GODAVARI  2001    23                        26   
...              ...            ...   ...   ...                       ...   
9012     LAKSHADWEEP    LAKSHADWEEP  2012     0                         0   
9013     LAKSHADWEEP          TOTAL  2012     0                         0   
9014      PUDUCHERRY       KARAIKAL  2012     6                         2   
9015      PUDUCHERRY     PUDUCHERRY  2012     7                        14   
9016      PUDUCHERRY          TOTAL  2012    13                        16   

      Dowry Deaths  Assault on women with intent to outrage her modesty  \
0               16                                                149     
1                7                                                118     
2               14                                                112     
3               17                                                126     
4               12                                                109     
...            ...                                                ...     
9012             0                                                  1     
9013             0                                                  1     
9014             0                                                  2     
9015             0                                                  7     
9016             0                                                  9     

      Insult to modesty of Women  Cruelty by Husband or his Relatives  \
0                             34                                  175   
1                             24                                  154   
2                             83                                  186   
3                             38                                   57   
4                             58                                  247   
...                          ...                                  ...   
9012                           0                                    1   
9013                           0                                    1   
9014                           0                                    1   
9015                           2                                    5   
9016                           2                                    6   

      Importation of Girls  
0                        0  
1                        0  
2                        0  
3                        0  
4                        0  
...                    ...  
9012                     0  
9013                     0  
9014                     0  
9015                     0  
9016                     0  

[9017 rows x 10 columns]

#### Total population of all states

In [15]:
df1 = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\Phase-1\all-csvs\All Year Population .csv")
df1

State/UT  Year  Population
0    Andaman and Nicobar Islands  2001      356152
1    Andaman and Nicobar Islands  2002      372096
2    Andaman and Nicobar Islands  2003      388343
3    Andaman and Nicobar Islands  2004      404902
4    Andaman and Nicobar Islands  2005      421779
..                           ...   ...         ...
427                  West Bengal  2008    87363895
428                  West Bengal  2009    88336987
429                  West Bengal  2010    89299017
430                  West Bengal  2011    90251375
431                  West Bengal  2012    91193022

[432 rows x 3 columns]

#### Literacy rates of all states

In [16]:
df2 = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\Phase-1\all-csvs\Literacy_Rate_2001_2012.csv")
df2

Unnamed: 0           State/UT   2001   2002   2003   2004   2005   2006  \
0            0      A & N ISLANDS  81.30  81.42  81.55  81.67  81.79  81.92   
1            1     Andhra Pradesh  60.47  62.66  64.84  66.11  67.39  68.68   
2            2  Arunachal Pradesh  54.34  56.94  59.54  62.15  64.75  67.36   
3            3              Assam  63.25  66.59  69.92  73.25  76.58  79.91   
4            4              Bihar  47.00  49.66  52.33  54.99  57.66  60.33   
5            5         Chandigarh  81.94  82.17  82.39  82.62  82.85  83.07   
6            6       Chhattisgarh  64.66  66.41  68.17  69.92  71.68  73.43   
7            7       D & N HAVELI  57.63  60.13  62.63  65.13  67.63  70.13   
8            8        DAMAN & DIU  78.18  80.68  83.18  85.68  88.18  90.68   
9            9              Delhi  81.67  81.79  81.92  82.04  82.17  82.29   
10          10                Goa  82.01  82.29  82.58  82.87  83.15  83.44   
11          11            Gujarat  69.14  70.96  72.79  74.61  76.44  78.26   
12          12            Haryana  67.91  68.81  69.70  70.60  71.49  72.39   
13          13   Himachal Pradesh  76.48  77.04  77.61  78.17  78.74  79.30   
14          14    JAMMU & KASHMIR  55.52  57.00  58.49  59.97  61.45  62.93   
15          15          Jharkhand  53.56  55.86  58.16  60.46  62.76  65.06   
16          16          Karnataka  66.06  67.55  69.04  70.53  72.02  73.51   
17          17             Kerala  90.86  91.06  91.26  91.46  91.66  91.86   
18          18        Lakshadweep  86.66  86.89  87.11  87.34  87.57  87.79   
19          19     Madhya Pradesh  63.74  64.47  65.20  65.93  66.66  67.39   
20          20        Maharashtra  76.84  77.65  78.46  79.27  80.08  80.89   
21          21            Manipur  70.50  71.45  72.41  73.37  74.33  75.28   
22          22          Meghalaya  62.56  63.35  64.14  64.94  65.73  66.52   
23          23            Mizoram  88.80  89.27  89.74  90.21  90.69  91.16   
24          24           Nagaland  66.59  68.48  70.36  72.25  74.14  76.02   
25          25             Odisha  63.08  64.49  65.90  67.32  68.73  70.14   
26          26         Puducherry  81.24  81.46  81.69  81.91  82.14  82.36   
27          27             Punjab  69.65  70.55  71.44  72.34  73.23  74.13   
28          28          Rajasthan  60.41  61.53  62.66  63.79  64.91  66.04   
29          29             Sikkim  68.81  69.42  70.04  70.65  71.27  71.88   
30          30         Tamil Nadu  73.45  74.12  74.79  75.46  76.14  76.81   
31          31            Tripura  73.19  74.35  75.51  76.68  77.84  79.00   
32          32      Uttar Pradesh  56.27  57.43  58.59  59.75  60.91  62.07   
33          33        Uttarakhand  71.62  72.09  72.56  73.03  73.50  73.97   
34          34        West Bengal  68.64  69.64  70.64  71.65  72.65  73.65   
35          35              India  64.84  65.74  66.64  67.55  68.45  69.35   

     2007   2008   2009   2010   2011   2012\t\t\t\t\t\t\t  
0   82.04  82.17  82.29  82.42  86.05  86.63\t\t\t\t\t\t\t  
1   69.97  71.26  72.55  73.84  67.02      -\t\t\t\t\t\t\t  
2   69.96  72.57  75.17  77.77  65.38      -\t\t\t\t\t\t\t  
3   83.24  86.57  89.90  93.23  72.19      -\t\t\t\t\t\t\t  
4   63.00  65.67  68.34  71.01  61.80      -\t\t\t\t\t\t\t  
5   83.30  83.53  83.75  83.98  86.05  86.05\t\t\t\t\t\t\t  
6   75.19  76.94  78.70  80.45  70.28      -\t\t\t\t\t\t\t  
7   72.63  75.13  77.63  80.13  76.24      -\t\t\t\t\t\t\t  
8   93.18  95.68  98.18      -  87.10      -\t\t\t\t\t\t\t  
9   82.42  82.55  82.67  82.80  86.21      -\t\t\t\t\t\t\t  
10  83.73  84.01  84.30  84.59  88.70      -\t\t\t\t\t\t\t  
11  80.09  81.91  83.74  85.56  78.03      -\t\t\t\t\t\t\t  
12  73.29  74.18  75.08  75.98  75.55      -\t\t\t\t\t\t\t  
13  79.87  80.43  81.00  81.56  82.80      -\t\t\t\t\t\t\t  
14  64.42  65.90  67.38  68.86  67.16      -\t\t\t\t\t\t\t  
15  67.36  69.66  71.96  74.26  66.41      -\t\t\t\t\t\t\t  
16  75.00  76.49  77.98  79.47  75.36      

#### Area covered of all states from 2001 to 2011

In [17]:
ac = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\Phase-1\all-csvs\All states area covered.csv")
ac

Unnamed: 0           State/UT  2001 Area (in sq. km)  \
0            0      A & N ISLANDS                   8249   
1            1     Andhra Pradesh                 275045   
2            2  Arunachal Pradesh                  83743   
3            3              Assam                  78438   
4            4              Bihar                  94163   
5            5         Chandigarh                    114   
6            6       Chhattisgarh                 135191   
7            7       D & N HAVELI                    491   
8            8        Daman & Diu                    112   
9            9              Delhi                   1483   
10          10                Goa                   3702   
11          11            Gujarat                 196024   
12          12            Haryana                  44212   
13          13   Himachal Pradesh                  55673   
14          14    Jammu & Kashmir                 222236   
15          15          Jharkhand                  74677   
16          16          Karnataka                 191791   
17          17             Kerala                  38852   
18          18        Lakshadweep                     32   
19          19     Madhya Pradesh                 308245   
20          20        Maharashtra                 307713   
21          21            Manipur                  22327   
22          22          Meghalaya                  22429   
23          23            Mizoram                  21081   
24          24           Nagaland                  16579   
25          25             Odisha                 155707   
26          26         Puducherry                    479   
27          27             Punjab                  50362   
28          28          Rajasthan                 342239   
29          29             Sikkim                   7096   
30          30         Tamil Nadu                 130060   
31          31            Tripura                  10486   
32          32      Uttar Pradesh                 240928   
33          33        Uttarakhand                  53566   
34          34        West Bengal                  88752   

    2002 Area (in sq. km)  2003 Area (in sq. km)  2004 Area (in sq. km)  \
0                    8249                   8249                   8249   
1                  275045                 275045                 275045   
2                   83743                  83743                  83743   
3                   78438                  78438                  78438   
4                   94163                  94163                  94163   
5                     114                    114                    114   
6                  135191                 135191                 135191   
7                     491                    491                    491   
8                     112                    112                    112   
9                    1483                   1483                   1483   
10                   3702                   3702                   3702   
11                 196024                 196024                 196024   
12                  44212                  44212                  44212   
13                  55673                  55673                  55673   
14                 222236                 222236                 222236   
15                  74677                  74677                  74677   
16                 191791                 191791                 191791   
17                  38852                  38852                  38852   
18                     32                     32                     32   
19                 308245                 308245                 308245   
20                 307713                 307713                 307713   
21                  22327                  22327                  22327   
22                  22429                  22429                  22429   
23                  21081                  

Now, let's combine all of the previously mentioned datasets into a single comprehensive megaset.

In [34]:
df = pd.read_csv(r"C:\Users\AbhilashKY\Capstone_project_Crime_rates_in_India\Phase-1\all-csvs\All_crimes_of_India_Crime .csv")
df

STATE/UT  YEAR         STATE AND YEAR  POPULATION  \
0       ANDHRA PRADESH  2001     ANDHRA PRADESH2001    76210007   
1    ARUNACHAL PRADESH  2001  ARUNACHAL PRADESH2001     1097968   
2                ASSAM  2001              ASSAM2001    26655528   
3                BIHAR  2001              BIHAR2001    82998509   
4         CHHATTISGARH  2001       CHHATTISGARH2001    20733232   
..                 ...   ...                    ...         ...   
403         CHANDIGARH  2012         CHANDIGARH2012     1144839   
404       D & N HAVELI  2012       D & N HAVELI2012      259148   
405        DAMAN & DIU  2012        DAMAN & DIU2012      191924   
406        LAKSHADWEEP  2012        LAKSHADWEEP2012       74578   
407         PUDUCHERRY  2012         PUDUCHERRY2012     1148900   

     LITERACY RATE  AREA (in sq. km)  MURDER  ATTEMPT TO MURDER  \
0            60.47          275045.0    2602               1555   
1            54.34           83743.0      83                 53   
2            63.25           78438.0    1356                481   
3            47.00           94163.0    3643               3419   
4            64.66          135191.0     880                529   
..             ...               ...     ...                ...   
403          86.05               NaN      21                 44   
404            NaN               NaN       7                  2   
405            NaN               NaN       3                  5   
406            NaN               NaN       0                  0   
407            NaN               NaN      29                 27   

     CULPABLE HOMICIDE NOT AMOUNTING TO MURDER  RAPE  ...  TOTAL  Dacoity  \
0                                          136   871  ...           214.0   
1                                            3    33  ...            22.0   
2                                           40   817  ...           532.0   
3                                          250   888  ...          1291.0   
4                                           45   959  ...            87.0   
..                                         ...   ...  ...             ...   
403                                          5    27  ...             3.0   
404                                          0     3  ...             2.0   
405                                          0     5  ...             1.0   
406                                          0     0  ...             0.0   
407                                         12    13  ...             5.0   

     TOTAL  Robbery  TOTAL  Burglary  TOTAL  Theft  Total Petty Crime  \
0             629.0           7220.0       16751.0            24814.0   
1              84.0            248.0         443.0              797.0   
2             687.0           2423.0        5367.0             9009.0   
3            2203.0           3233.0        9701.0            16428.0   
4             338.0           4144.0        4812.0             9381.0   
..              ...              ...           ...                ...   
403            63.0            235.0        1549.0             1850.0   
404             2.0             22.0          68.0               94.0   
405             3.0             26.0          55.0               85.0   
406             0.0              3.0           7.0               10.0   
407            16.0             77.0         658.0              756.0   

     Human Rights Violation Cases  Policemen Chargesheeted  Polimen Charged  \
0                            45.0                     29.0              2.0   
1                            11.0                      6.0              0.0   
2                            35.0                      0.0              0.0   
3                             0.0                      0.0              0.0   
4                             0.0                      0.0              0.0   
..                            ...                      ...              ...   
403                           NaN                

Let's start analysing the combined dataset

In [35]:
# Checking shape
df.shape

(408, 52)

In [36]:
# Brief summary of our dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 408 entries, 0 to 407
Data columns (total 52 columns):
 #   Column                                               Non-Null Count  Dtype  
---  ------                                               --------------  -----  
 0   STATE/UT                                             408 non-null    object 
 1   YEAR                                                 408 non-null    int64  
 2   STATE AND YEAR                                       408 non-null    object 
 3   POPULATION                                           408 non-null    int64  
 4   LITERACY RATE                                        375 non-null    float64
 5   AREA (in sq. km)                                     374 non-null    float64
 6   MURDER                                               408 non-null    int64  
 7   ATTEMPT TO MURDER                                    408 non-null    int64  
 8   CULPABLE HOMICIDE NOT AMOUNTING TO MURDER            408 non-null    i

#### Observations:
The info method gives us the summary of our dataset as listed below:
1. Dataset contains 408 rows and 52 columns.
2. Few columns of dataset contain null values. We shall clean them using imputers in later stage
3. There are 14 columns as float type data, 37 columns as int data type and 1 column as object datatype
4. There is column called unnamed which can be removed since it is index and do not affect our data analysing
5. The memory used - 165.9+ KB

In [43]:
df.isnull().sum()

STATE/UT                                                 0
YEAR                                                     0
POPULATION                                               0
LITERACY RATE                                           33
AREA (in sq. km)                                        34
MURDER                                                   0
ATTEMPT TO MURDER                                        0
CULPABLE HOMICIDE NOT AMOUNTING TO MURDER                0
RAPE                                                     0
CUSTODIAL RAPE                                           0
OTHER RAPE                                               0
KIDNAPPING & ABDUCTION                                   0
KIDNAPPING AND ABDUCTION OF WOMEN AND GIRLS              0
KIDNAPPING AND ABDUCTION OF OTHERS                       0
DACOITY                                                  0
PREPARATION AND ASSEMBLY FOR DACOITY                     0
ROBBERY                                                 

There are null values in our dataset as seen above

Filling all null values from the dataset using iterative imputer

In [44]:

from sklearn.experimental import enable_iterative_imputer
from sklearn.impute import IterativeImputer

In [45]:
it = IterativeImputer()

In [38]:
df.columns

Index(['STATE/UT', 'YEAR', 'STATE AND YEAR', 'POPULATION', 'LITERACY RATE',
       'AREA (in sq. km)', 'MURDER', 'ATTEMPT TO MURDER',
       'CULPABLE HOMICIDE NOT AMOUNTING TO MURDER', 'RAPE', 'CUSTODIAL RAPE',
       'OTHER RAPE', 'KIDNAPPING & ABDUCTION',
       'KIDNAPPING AND ABDUCTION OF WOMEN AND GIRLS',
       'KIDNAPPING AND ABDUCTION OF OTHERS', 'DACOITY',
       'PREPARATION AND ASSEMBLY FOR DACOITY', 'ROBBERY', 'BURGLARY', 'THEFT',
       'AUTO THEFT', 'OTHER THEFT', 'RIOTS', 'CRIMINAL BREACH OF TRUST',
       'CHEATING', 'COUNTERFIETING', 'ARSON', 'HURT/GREVIOUS HURT',
       'DOWRY DEATHS', 'ASSAULT ON WOMEN WITH INTENT TO OUTRAGE HER MODESTY',
       'INSULT TO MODESTY OF WOMEN', 'CRUELTY BY HUSBAND OR HIS RELATIVES',
       'IMPORTATION OF GIRLS FROM FOREIGN COUNTRIES',
       'CAUSING DEATH BY NEGLIGENCE', 'OTHER IPC CRIMES', 'CRIMES AGAINST SC',
       'CRIMES AGAINST ST', 'CRIMES AGAINST CHILDREN', 'JUVENILES APPREHENDED',
       'POLICEMEN KILLED', 'THEFT CASES REPO

In [40]:
#Dropping columns which are irrelevant such as STATE AND YEAR
df = df.drop('STATE AND YEAR', axis =1)
df


STATE/UT  YEAR  POPULATION  LITERACY RATE  AREA (in sq. km)  \
0       ANDHRA PRADESH  2001    76210007          60.47          275045.0   
1    ARUNACHAL PRADESH  2001     1097968          54.34           83743.0   
2                ASSAM  2001    26655528          63.25           78438.0   
3                BIHAR  2001    82998509          47.00           94163.0   
4         CHHATTISGARH  2001    20733232          64.66          135191.0   
..                 ...   ...         ...            ...               ...   
403         CHANDIGARH  2012     1144839          86.05               NaN   
404       D & N HAVELI  2012      259148            NaN               NaN   
405        DAMAN & DIU  2012      191924            NaN               NaN   
406        LAKSHADWEEP  2012       74578            NaN               NaN   
407         PUDUCHERRY  2012     1148900            NaN               NaN   

     MURDER  ATTEMPT TO MURDER  CULPABLE HOMICIDE NOT AMOUNTING TO MURDER  \
0      2602               1555                                        136   
1        83                 53                                          3   
2      1356                481                                         40   
3      3643               3419                                        250   
4       880                529                                         45   
..      ...                ...                                        ...   
403      21                 44                                          5   
404       7                  2                                          0   
405       3                  5                                          0   
406       0                  0                                          0   
407      29                 27                                         12   

     RAPE  CUSTODIAL RAPE  ...  TOTAL  Dacoity  TOTAL  Robbery  \
0     871               0  ...           214.0           629.0   
1      33               0  ...            22.0            84.0   
2     817               0  ...           532.0           687.0   
3     888               0  ...          1291.0          2203.0   
4     959               0  ...            87.0           338.0   
..    ...             ...  ...             ...             ...   
403    27               0  ...             3.0            63.0   
404     3               0  ...             2.0             2.0   
405     5               0  ...             1.0             3.0   
406     0               0  ...             0.0             0.0   
407    13               0  ...             5.0            16.0   

     TOTAL  Burglary  TOTAL  Theft  Total Petty Crime  \
0             7220.0       16751.0            24814.0   
1              248.0         443.0              797.0   
2             2423.0        5367.0             9009.0   
3             3233.0        9701.0            16428.0   
4             4144.0        4812.0             9381.0   
..               ...           ...                ...   
403            235.0        1549.0             1850.0   
404             22.0          68.0               94.0   
405             26.0          55.0               85.0   
406              3.0           7.0               10.0   
407             77.0         658.0              756.0   

     Human Rights Violation Cases  Policemen Chargesheeted  Polimen Charged  \
0                            45.0                     29.0              2.0   
1                            11.0                      6.0              0.0   
2                            35.0                      0.0              0.0   
3                             0.0                      0.0              0.0   
4                             0.0                      0.0              0.0   
..                            ...                      ...              ...   
403                           NaN                      NaN              NaN   
404                           NaN                     

In [46]:
iter_col = pd.DataFrame(it.fit_transform(df[['LITERACY RATE', 'AREA (in sq. km)','CRIMES AGAINST CHILDREN', 'JUVENILES APPREHENDED','POLICEMEN KILLED', 'THEFT CASES REPOERTED', 'THEFT VALUE TOTAL','TOTAL  Dacoity', 'TOTAL  Robbery', 'TOTAL  Burglary', 'TOTAL  Theft', 'Total Petty Crime', 'Human Rights Violation Cases', 'Policemen Chargesheeted', 'Polimen Charged']]),
                        columns = ['LITERACY RATE_ITER', 'AREA (in sq. km)_ITER','CRIMES AGAINST CHILDREN_ITER', 'JUVENILES APPREHENDED_ITER','POLICEMEN KILLED_ITER', 'THEFT CASES REPOERTED_ITER', 'THEFT VALUE TOTAL_ITER','TOTAL  Dacoity_ITER', 'TOTAL  Robbery_ITER', 'TOTAL  Burglary_ITER', 'TOTAL  Theft_ITER', 'Total Petty Crime_ITER', 'Human Rights Violation Cases_ITER', 'Policemen Chargesheeted_ITER', 'Polimen Charged_ITER'])

df = df.join(iter_col)
df

STATE/UT  YEAR  POPULATION  LITERACY RATE  AREA (in sq. km)  \
0       ANDHRA PRADESH  2001    76210007          60.47          275045.0   
1    ARUNACHAL PRADESH  2001     1097968          54.34           83743.0   
2                ASSAM  2001    26655528          63.25           78438.0   
3                BIHAR  2001    82998509          47.00           94163.0   
4         CHHATTISGARH  2001    20733232          64.66          135191.0   
..                 ...   ...         ...            ...               ...   
403         CHANDIGARH  2012     1144839          86.05               NaN   
404       D & N HAVELI  2012      259148            NaN               NaN   
405        DAMAN & DIU  2012      191924            NaN               NaN   
406        LAKSHADWEEP  2012       74578            NaN               NaN   
407         PUDUCHERRY  2012     1148900            NaN               NaN   

     MURDER  ATTEMPT TO MURDER  CULPABLE HOMICIDE NOT AMOUNTING TO MURDER  \
0      2602               1555                                        136   
1        83                 53                                          3   
2      1356                481                                         40   
3      3643               3419                                        250   
4       880                529                                         45   
..      ...                ...                                        ...   
403      21                 44                                          5   
404       7                  2                                          0   
405       3                  5                                          0   
406       0                  0                                          0   
407      29                 27                                         12   

     RAPE  CUSTODIAL RAPE  ...  THEFT CASES REPOERTED_ITER  \
0     871               0  ...                24814.000000   
1      33               0  ...                  797.000000   
2     817               0  ...                 9009.000000   
3     888               0  ...                16428.000000   
4     959               0  ...                26923.309353   
..    ...             ...  ...                         ...   
403    27               0  ...                 1850.000000   
404     3               0  ...                   94.000000   
405     5               0  ...                   85.000000   
406     0               0  ...                   10.000000   
407    13               0  ...                  756.000000   

     THEFT VALUE TOTAL_ITER  TOTAL  Dacoity_ITER  TOTAL  Robbery_ITER  \
0              4.231238e+08                214.0                629.0   
1              2.583082e+07                 22.0                 84.0   
2              9.193986e+07                532.0                687.0   
3              4.035582e+08               1291.0               2203.0   
4              3.247267e+08                 87.0                338.0   
..                      ...                  ...                  ...   
403            8.347295e+07                  3.0                 63.0   
404            1.383436e+07                  2.0                  2.0   
405            1.528780e+07                  1.0                  3.0   
406            2.799750e+05                  0.0                  0.0   
407            3.923804e+07                  5.0                 16.0   

     TOTAL  Burglary_ITER  TOTAL  Theft_ITER  Total Petty Crime_ITER  \
0                  7220.0            16751.0                 24814.0   
1                   248.0              443.0                   797.0   
2                  2423.0             5367.0                  9009.0   
3                  3233.0             9701.0                 16428.0   
4                  4144.0             4812.0                  9381.0   
..                    ...                ...                     ...   
403                 235.0             

In [47]:
#Dropping columns which are repetative
drop_columns = ['LITERACY RATE', 'AREA (in sq. km)','CRIMES AGAINST CHILDREN', 'JUVENILES APPREHENDED','POLICEMEN KILLED', 'THEFT CASES REPOERTED', 'THEFT VALUE TOTAL','TOTAL  Dacoity', 'TOTAL  Robbery', 'TOTAL  Burglary', 'TOTAL  Theft', 'Total Petty Crime', 'Human Rights Violation Cases', 'Policemen Chargesheeted', 'Polimen Charged']

In [48]:
df = df.drop(columns = drop_columns, axis=1)
df

STATE/UT  YEAR  POPULATION  MURDER  ATTEMPT TO MURDER  \
0       ANDHRA PRADESH  2001    76210007    2602               1555   
1    ARUNACHAL PRADESH  2001     1097968      83                 53   
2                ASSAM  2001    26655528    1356                481   
3                BIHAR  2001    82998509    3643               3419   
4         CHHATTISGARH  2001    20733232     880                529   
..                 ...   ...         ...     ...                ...   
403         CHANDIGARH  2012     1144839      21                 44   
404       D & N HAVELI  2012      259148       7                  2   
405        DAMAN & DIU  2012      191924       3                  5   
406        LAKSHADWEEP  2012       74578       0                  0   
407         PUDUCHERRY  2012     1148900      29                 27   

     CULPABLE HOMICIDE NOT AMOUNTING TO MURDER  RAPE  CUSTODIAL RAPE  \
0                                          136   871               0   
1                                            3    33               0   
2                                           40   817               0   
3                                          250   888               0   
4                                           45   959               0   
..                                         ...   ...             ...   
403                                          5    27               0   
404                                          0     3               0   
405                                          0     5               0   
406                                          0     0               0   
407                                         12    13               0   

     OTHER RAPE  KIDNAPPING & ABDUCTION  ...  THEFT CASES REPOERTED_ITER  \
0           871                    1182  ...                24814.000000   
1            33                      83  ...                  797.000000   
2           817                    1480  ...                 9009.000000   
3           888                    2159  ...                16428.000000   
4           959                     207  ...                26923.309353   
..          ...                     ...  ...                         ...   
403          27                      87  ...                 1850.000000   
404           3                      13  ...                   94.000000   
405           5                       3  ...                   85.000000   
406           0                       0  ...                   10.000000   
407          13                      19  ...                  756.000000   

     THEFT VALUE TOTAL_ITER  TOTAL  Dacoity_ITER  TOTAL  Robbery_ITER  \
0              4.231238e+08                214.0                629.0   
1              2.583082e+07                 22.0                 84.0   
2              9.193986e+07                532.0                687.0   
3              4.035582e+08               1291.0               2203.0   
4              3.247267e+08                 87.0                338.0   
..                      ...                  ...                  ...   
403            8.347295e+07                  3.0                 63.0   
404            1.383436e+07                  2.0                  2.0   
405            1.528780e+07                  1.0                  3.0   
406            2.799750e+05                  0.0                  0.0   
407            3.923804e+07                  5.0                 16.0   

     TOTAL  Burglary_ITER  TOTAL  Theft_ITER  Total Petty Crime_ITER  \
0                  7220.0            16751.0                 24814.0   
1                   248.0              443.0                   797.0   
2                  2423.0             5367.0                  9009.0   
3                  3233.0             9701.0                 16428.0   
4                  4144.0             4812.0                  9381.0   
..                    ...                ...                     ...   
40

In [49]:
df.isnull().sum()

STATE/UT                                               0
YEAR                                                   0
POPULATION                                             0
MURDER                                                 0
ATTEMPT TO MURDER                                      0
CULPABLE HOMICIDE NOT AMOUNTING TO MURDER              0
RAPE                                                   0
CUSTODIAL RAPE                                         0
OTHER RAPE                                             0
KIDNAPPING & ABDUCTION                                 0
KIDNAPPING AND ABDUCTION OF WOMEN AND GIRLS            0
KIDNAPPING AND ABDUCTION OF OTHERS                     0
DACOITY                                                0
PREPARATION AND ASSEMBLY FOR DACOITY                   0
ROBBERY                                                0
BURGLARY                                               0
THEFT                                                  0
AUTO THEFT                     

All null values have been filled

In [51]:
df.describe().T

count          mean  \
YEAR                                                408.0  2.006500e+03   
POPULATION                                          408.0  3.257161e+07   
MURDER                                              408.0  9.714167e+02   
ATTEMPT TO MURDER                                   408.0  8.497549e+02   
CULPABLE HOMICIDE NOT AMOUNTING TO MURDER           408.0  1.071838e+02   
RAPE                                                408.0  5.703750e+02   
CUSTODIAL RAPE                                      408.0  6.372549e-02   
OTHER RAPE                                          408.0  5.703113e+02   
KIDNAPPING & ABDUCTION                              408.0  8.131716e+02   
KIDNAPPING AND ABDUCTION OF WOMEN AND GIRLS         408.0  6.084730e+02   
KIDNAPPING AND ABDUCTION OF OTHERS                  408.0  2.046985e+02   
DACOITY                                             408.0  1.446765e+02   
PREPARATION AND ASSEMBLY FOR DACOITY                408.0  7.524510e+01   
ROBBERY                                             408.0  5.926618e+02   
BURGLARY                                            408.0  2.680843e+03   
THEFT                                               408.0  8.012216e+03   
AUTO THEFT                                          408.0  2.802336e+03   
OTHER THEFT                                         408.0  5.209880e+03   
RIOTS                                               408.0  1.896890e+03   
CRIMINAL BREACH OF TRUST                            408.0  4.394510e+02   
CHEATING                                            408.0  1.817326e+03   
COUNTERFIETING                                      408.0  6.421078e+01   
ARSON                                               408.0  2.773039e+02   
HURT/GREVIOUS HURT                                  408.0  8.209966e+03   
DOWRY DEATHS                                        408.0  2.196569e+02   
ASSAULT ON WOMEN WITH INTENT TO OUTRAGE HER MOD...  408.0  1.092208e+03   
INSULT TO MODESTY OF WOMEN                          408.0  2.999191e+02   
CRUELTY BY HUSBAND OR HIS RELATIVES                 408.0  2.107819e+03   
IMPORTATION OF GIRLS FROM FOREIGN COUNTRIES         408.0  2.183824e+00   
CAUSING DEATH BY NEGLIGENCE                         408.0  2.437159e+03   
OTHER IPC CRIMES                                    408.0  2.344454e+04   
CRIMES AGAINST SC                                   408.0  9.067647e+02   
CRIMES AGAINST ST                                   408.0  1.704167e+02   
Crimes Against Women                                408.0  4.900635e+03   
TOTAL IPC CRIMES                                    408.0  5.712618e+04   
LITERACY RATE_ITER                                  408.0  8.460836e+01   
AREA (in sq. km)_ITER                               408.0  1.238004e+05   
CRIMES AGAINST CHILDREN_ITER                        408.0  4.189257e+28   
JUVENILES APPREHENDED_ITER                          408.0  9.919854e+02   
POLICEMEN KILLED_ITER                               408.0 -7.591898e+00   
THEFT CASES REPOERTED_ITER                          408.0 -8.149591e+36   
THEFT VALUE TOTAL_ITER                              408.0 -7.227924e+40   
TOTAL  Dacoity_ITER                                 408.0 -1.317265e+19   
TOTAL  Robbery_ITER                                 408.0 -5.919376e+19   
TOTAL  Burglary_ITER                                408.0 -1.968673e+20   
TOTAL  Theft_ITER                                   408.0 -5.875258e+20   
Total Petty Crime_ITER                              408.0 -2.726635e+21   
Human Rights Violation Cases_ITER                   408.0 -2.062519e+27   
Policemen Chargesheeted_ITER                        408.0  2.469295e+27   
Polimen Charged_ITER                                408.0 -6.385597e+25   

                                                             std  \
YEAR                                                3.456291e+00   
POPULATION                                          4.130209e+07   
MURDER                

In [55]:
# Expandas exponential values
pd.options.display.float_format = '{:.6f}'.format

In [56]:
df.describe()

YEAR       POPULATION      MURDER  ATTEMPT TO MURDER  \
count  408.000000       408.000000  408.000000         408.000000   
mean  2006.500000  32571614.355392  971.416667         849.754902   
std      3.456291  41302087.011694 1229.031743        1164.326125   
min   2001.000000     60650.000000    0.000000           0.000000   
25%   2003.750000   1224418.500000   45.000000          29.750000   
50%   2006.500000  16229848.000000  381.000000         432.500000   
75%   2009.250000  57400463.750000 1479.750000        1407.500000   
max   2012.000000 205888020.000000 7601.000000        7964.000000   

       CULPABLE HOMICIDE NOT AMOUNTING TO MURDER        RAPE  CUSTODIAL RAPE  \
count                                 408.000000  408.000000      408.000000   
mean                                  107.183824  570.375000        0.063725   
std                                   257.514272  704.930637        0.397607   
min                                     0.000000    0.000000        0.000000   
25%                                     4.000000   26.750000        0.000000   
50%                                    28.000000  281.500000        0.000000   
75%                                    94.000000  961.500000        0.000000   
max                                  1616.000000 3425.000000        5.000000   

       OTHER RAPE  KIDNAPPING & ABDUCTION  \
count  408.000000              408.000000   
mean   570.311275              813.171569   
std    704.883153             1204.731087   
min      0.000000                0.000000   
25%     26.750000               32.750000   
50%    281.500000              267.000000   
75%    961.500000             1081.000000   
max   3425.000000             8878.000000   

       KIDNAPPING AND ABDUCTION OF WOMEN AND GIRLS  ...  \
count                                   408.000000  ...   
mean                                    608.473039  ...   
std                                     970.140758  ...   
min                                       0.000000  ...   
25%                                      14.000000  ...   
50%                                     196.500000  ...   
75%                                     828.750000  ...   
max                                    7910.000000  ...   

                            THEFT CASES REPOERTED_ITER  \
count                                       408.000000   
mean     -8149590948941492896162451437903675392.000000   
std     164613668068929498671308396782195572736.000000   
min   -3325033107168129167747410946839732551680.000000   
25%                                       -1748.690647   
50%                                        2347.309353   
75%                                       22295.500000   
max                                      354603.309353   

                                  THEFT VALUE TOTAL_ITER  \
count                                         408.000000   
mean   -72279235088189217227077341862326381314048.000000   
std   1459968983428400582133346010061577549512704.000000   
min   -29489927915981201015503817756510499482107904.0...   
25%                                       2262655.697614   
50%                                     222545800.511430   
75%                                     698526221.684729   
max                                    5157062253.000000   

                 TOTAL  Dacoity_ITER             TOTAL  Robbery_ITER  \
count                     408.000000                      408.000000   
mean    -13172648926884161536.000000    -59193755369594978304.000000   
std     266074465776753344512.000000   1195655249350829015040.000000   
min   -5374440762168737529856.000000 -24151052190794752458752.000000   
25%                         4.000000                       18.000000   
50%                        42.000000                      162.500000   
75%                       199.000000                      783.500000   
max                      1319.000000                     6949.000000   

                 TO

The describe() method gives us Stastical Summary of our dataset 

We have done preliminary changes to our dataset and cleaned for the later stages of analysis

In [58]:
file_path = 'Final_Dataset__of_India_Crimes.csv'

In [59]:
df.to_csv(file_path, index=False)

The dataset has been saved as csv file